freqtrade/exchange/exchange.py

```diff
@@ -1983,6 +1982,126 @@ def get_funding_fees(
+            raise OperationalException(f'{self.name}.collateral must be set for liquidation_price')
+        elif (self.trading_mode != TradingMode.FUTURES and self.collateral != Collateral.ISOLATED):
+            raise OperationalException(
+                f"{self.name} does not support {self.collateral.value} {self.trading_mode.value}")
+
+        if self._config['dry_run'] or not self.exchange_has("fetchPositions"):
+            if (
+                open_rate is None or
+                is_short is None or
+                position is None or
+                wallet_balance is None
+            ):
+                raise OperationalException(
+                    f"Parameters open_rate, is_short, position, wallet_balance are"
+                    f"required by {self.name}.liquidation_price for dry_run"
+                )
+
+            return self.dry_run_liquidation_price(
+                pair=pair,
+                open_rate=open_rate,
+                is_short=is_short,
+                position=position,
+                wallet_balance=wallet_balance,
+                mm_ex_1=mm_ex_1,
+                upnl_ex_1=upnl_ex_1
+            )
+
+        try:
+            positions = self._api.fetch_positions([pair])
+            pos = positions[0]
+            return pos['liquidationPrice']
+        except ccxt.DDoSProtection as e:
+            raise DDosProtection(e) from e
+        except (ccxt.NetworkError, ccxt.ExchangeError) as e:
+            raise TemporaryError(
+                f'Could not set margin mode due to {e.__class__.__name__}. Message: {e}') from e
+        except ccxt.BaseError as e:
+            raise OperationalException(e) from e
+
+    def get_maintenance_ratio_and_amt(
+        self,
+        pair: str,
+        nominal_value: Optional[float] = 0.0,
+    ) -> Tuple[float, Optional[float]]:
+        """
+        :return: The maintenance margin ratio and maintenance amount
+        """
+        raise OperationalException(self.name + ' does not support leverage futures trading')
+
+    def dry_run_liquidation_price(
+        self,
+        pair: str,
+        open_rate: float,   # Entry price of position
+        is_short: bool,
+        position: float,  # Absolute value of position size
+        wallet_balance: float,  # Or margin balance
+        mm_ex_1: float = 0.0,  # (Binance) Cross only
+        upnl_ex_1: float = 0.0,  # (Binance) Cross only
+    ) -> Optional[float]:
+        """
+        PERPETUAL:
+         gateio: https://www.gate.io/help/futures/perpetual/22160/calculation-of-liquidation-price
+         okex: https://www.okex.com/support/hc/en-us/articles/
+            360053909592-VI-Introduction-to-the-isolated-mode-of-Single-Multi-currency-Portfolio-margin
+
+        :param exchange_name:
+        :param open_rate: Entry price of position
+        :param is_short: True if the trade is a short, false otherwise
+        :param position: Absolute value of position size (in base currency)
```

Comment:
Can you maybe point out if this is before or after applying leverage?
so is it 50, or 150 if i trade with 3x leverage?

Reply:
150, it's the value of your position